[Term Entry] PyTorch Tensor Operations: .is_tensor() (#6118)
Codecademy/docs: content/pytorch/concepts/tensor-operations/terms/is-tensor/is-tensor.md

---
Title: '.is_tensor()'
Description: 'Checks if the given object is a PyTorch tensor.'
Subjects:
  - 'AI'
  - 'Data Science'
Tags:
  - 'AI'
  - 'Arrays'
  - 'Data Structures'
  - 'Deep Learning'
CatalogContent:
  - 'intro-to-py-torch-and-neural-networks'
  - 'paths/computer-science'
---

> In PyTorch, the **`.is_tensor()`** function checks if the given object is a PyTorch [tensor](https://www.codecademy.com/resources/docs/pytorch/tensors) and returns `True` if it is. This function is part of the `torch` module.

Reviewer: This can be better written as:

```suggestion
In PyTorch, the **`.is_tensor()`** function checks if the given object is a PyTorch [tensor](https://www.codecademy.com/resources/docs/pytorch/tensors). It returns `True` if the object is a tensor; otherwise, it returns `False`. This function is part of the `torch` module.
```

## Syntax

```pseudo

> torch.is_tensor(obj, /)

Reviewer: ```suggestion
torch.is_tensor(obj)
```

```

> - `obj`: The input object to be tested.

Reviewer: ```suggestion
- `obj`: The object to check whether it is a PyTorch tensor.
```

## Example

The following example demonstrates the usage of the `.is_tensor()` function:

```py

> import torch
> 
> # Define an object
> obj = torch.tensor([11, 22, 33, 44, 55])
> 
> # Check if the given object is a PyTorch tensor
> res = torch.is_tensor(obj)
> 
> # Print the result
> print(res)

Reviewer: We should show both the cases -

```suggestion
import torch

# Define objects
tensor_obj = torch.tensor([11, 22, 33, 44, 55])  # Tensor object
list_obj = [11, 22, 33, 44, 55]  # Non-tensor object

# Check if the objects are PyTorch tensors
print(torch.is_tensor(tensor_obj))  # True
print(torch.is_tensor(list_obj))    # False
```

```

The above code produces the following output:

```shell

> True

Reviewer: The output will change based on above code changes:

```suggestion
True
False
```
